### PR TITLE
Migrate YAML input asset validation to validate-skill command

### LINKED ISSUE #32: The `dart test`s shouldn't be responsible for validating the input
The `dart test`s shouldn't be responsible for validating the input.  I have been trying to clean this up, but there are still some cases, see `./tool/test/yaml_assets_test.dart`, while okay in the short term to get bootstrapped, ideally, these kinds of checks would happen in a the validate command.

FI @johnpryan

### PR DESCRIPTION
**Pull request by jwren**
- Resolves https://github.com/flutter/skills/issues/32
- Move structural, prefix, and protocol checks from `yaml_assets_test.dart` to `ValidateSkillCommand.run()`.
- Allow configuration validation to run upfront without requiring a Gemini API key.
- Delete legacy `test/yaml_assets_test.dart`.
- Add comprehensive edge case coverage in `test/commands/validate_skills_input_test.dart`.

### CODE REVIEW

File: tool/generator/lib/src/commands/validate_skill_command.dart [hunk 1]
```diff
@@ -32,6 +33,127 @@ class ValidateSkillCommand extends BaseSkillCommand {
...
+            isValid = false;
+          }
+        } else if (fileName == 'dart_skills.yaml') {
+          if (!name.startsWith('dart-')) {
+            logger.severe(
+              'Skill name "$name" in dart_skills.yaml must start with "dart-".',
+            );
+            isValid = false;
+          }
+        }
```
[the remark this answers is not in the document]
Comment (jwren) [reply tool/generator/lib/src/commands/validate_skill_command.dart:139]: fixed

File: tool/generator/lib/src/commands/validate_skill_command.dart [hunk 2]
```diff
@@ -32,6 +33,127 @@ class ValidateSkillCommand extends BaseSkillCommand {
...
+        isValid = false;
+      }
+      if (item['description'] == null) {
+        logger.severe('Skill "$name" is missing required field "description".');
+        isValid = false;
+      }
+      if (item['resources'] == null) {
+        logger.severe('Skill "$name" is missing required field "resources".');
+        isValid = false;
+      }
```
[the remark this answers is not in the document]
Comment (jwren) [reply tool/generator/lib/src/commands/validate_skill_command.dart]: fixed

**Review by johnpryan — APPROVED** [review]
